enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/discovery/HazelcastClient.java

```diff
@@ -78,6 +83,19 @@
         this.refreshPeriod = config.get( CausalClusteringSettings.cluster_topology_refresh ).toMillis();
         this.myself = myself;
         this.groups = config.get( CausalClusteringSettings.server_groups );
+        this.topologyServiceRetryStrategy = resolveStrategy();
+    }
+
+    private TopologyServiceRetryStrategy resolveStrategy()
+    {
+        Duration defaultDuration = DURATION.apply( CausalClusteringSettings.cluster_topology_refresh.getDefaultValue() );
+        long delayMillis =
+                config.getValue( CausalClusteringSettings.cluster_topology_refresh.name() )
+                        .map( configValueToCast -> (Duration) configValueToCast)
+                        .orElse( defaultDuration )
+                        .toMillis() / 2;
```

Comment:
The default refresh rate is every 5s, so by dividing by 2 we can shorten the delay between polls while also keeping it within the refresh window. It is also the reason for a hard-coded "3 retries"